agents/service.js

```diff
@@ -60,6 +65,44 @@ const hooks = {
   error: {}
 }
 
+function agentAndCredentialAlreadyExist (hook) {
+  if (hook.data.type === 'group') return hook
+  const credentialsService = hook.app.service('credentials')
+  const agentsService = hook.app.service('agents')
+  const { contextAgentId, relationships, credential } = hook.params
+
+  if (isNil(credential)) return hook
+
+  return credentialsService.find({
+    query: {
+      email: credential.email
+    }
+  })
+  .then((credentialResults) => {
+    const agentIds = getAgentIds(credentialResults)
```

Comment:
there can only be one agent possible here right? IMO it'd be clearer if we grabbed the single agent out of the array here rather than on L94, thoughts @gregorykan ?

Reply:
hmm i think i like the current pattern because we'd always be guaranteed an array type until the end? also leaving open the possibility that we could have multiple credentials per agent :o

Reply:
hmmm not sure i see it but totally happy to merge this and leave it 😆